chemprop/uncertainty/predictor.py

```diff
@@ -91,11 +106,74 @@ def _calc_prediction_uncertainty(
 
 @UncertaintyPredictorRegistry.register("dropout")
 class DropoutPredictor(UncertaintyPredictor):
-    def _calc_prediction_uncertainty(
-        self, dataloader: DataLoader, models: Iterable[MPNN], trainer: pl.Trainer
-    ) -> tuple[Tensor, Tensor]:
-        ...
-        return
+    """
+    A :class:`DropoutPredictor` creates a virtual ensemble of via Monte Carlo dropout with the provided models [1]_.
+
+    References
+    -----------
+    .. [1] arXiv:1506.02142 [stat.ML]
+
+    Parameters
+    ----------
+    ensemble_size: int
+        The number of samples to draw for the ensemble.
+    dropout: float | None
+        The probability of dropping out units in the dropout layers. If unspecified,
+        dropout will be performed using training probability.
+    """
+
+    def __init__(self, ensemble_size: int, dropout : None | float = None):
+        self.ensemble_size = ensemble_size
+        self.dropout = dropout
+
+    def _calc_prediction_uncertainty(self, dataloader, models, trainer) -> Tensor:
+        if len(models) != 1:
+            raise ValueError(
+                "Dropout method for uncertainty only takes exactly one model."
+            )
+        model = next(iter(models))
+        self._setup_predict_wrapper(model)
+        individual_preds = []
+
+        for _ in range(self.ensemble_size):
+            predss = trainer.predict(model, dataloader)
+            preds = torch.concat(predss, 0)
+            individual_preds.append(preds)
+
+        stacked_preds = torch.stack(individual_preds, dim=0).float()
+        means = torch.mean(stacked_preds, dim=0).unsqueeze(-1)
```

Comment:
```suggestion
        means = torch.mean(stacked_preds, dim=0).unsqueeze(0)
```
The shape of `stacked_preds` is `m x n x t` where ``m`` is the number of models, ``n`` is the number of inputs, ``t`` is the number of tasks. The`means` should be returned in shape of `1 x n x t` to match the design you used in `EnsemblePredictor`. (see more in the PR #986)